app/lib/account/backend.dart

```diff
@@ -111,6 +112,59 @@ Future<User> requireAuthenticatedUser({AuthSource? source}) async {
   return user;
 }
 
+/// Holds the authenticated agent information.
+class AuthenticatedAgent {
```

Comment:
Can we say what an `AuthenticatedAgent` is`

```suggestion
/// An [AuthenticatedAgent] represents a _service_ that has been authenticated
/// and which may be allowed to operate on specific resources on pub.dev
///
/// Examples:
///  * A GCP service account may authenticate using an OIDC `id_token`,
///  * A Github Action may authenticate using an OIDC `id_token`.
class AuthenticatedAgent {
```

Reply:
Would the term `AuthenticatedService` be better, or `AuthenticateBot` since these are always bots or services.

I think `AuthenticateService` might be best since this really is intended to be _service accounts_ and github actions.